tests/test_pydough_functions/udf_pydough_functions.py

```diff
@@ -0,0 +1,147 @@
+            PERCENTAGE(EPSILON(order_info.total_price, order_info.global_avg, 10)), 4
+        ),
+        pct_e100=ROUND(
+            PERCENTAGE(EPSILON(order_info.total_price, order_info.global_avg, 100)), 4
+        ),
+        pct_e1000=ROUND(
+            PERCENTAGE(EPSILON(order_info.total_price, order_info.global_avg, 1000)), 4
+        ),
+        pct_e10000=ROUND(
+            PERCENTAGE(EPSILON(order_info.total_price, order_info.global_avg, 10000)), 4
+        ),
+    )
+
+
+def sqlite_udf_covar_pop():
+    order_info = (
+        nations.customers.CALCULATE(account_balance)
+        .WHERE(market_segment == "BUILDING")
+        .orders.WHERE((YEAR(order_date) == 1998) & (order_priority == "2-HIGH"))
+    )
+    return regions.CALCULATE(
+        region_name=name,
+        cvp_ab_otp=ROUND(
+            POPULATION_COVARIANCE(
+                order_info.account_balance, order_info.total_price / 1_000_000.0
+            ),
+            3,
+        ),
+    ).ORDER_BY(region_name)
+
+
+def sqlite_udf_nval():
+    return (
+        regions.CALCULATE(rname=name)
+        .nations.CALCULATE(
+            rname,
+            nname=name,
+            v1=NVAL(name, 3, by=name),
+            v2=NVAL(name, 1, by=name, per="regions"),
+            v3=NVAL(name, 2, by=name, per="regions", frame=(1, None)),
+            v4=NVAL(name, 5, by=name, cumulative=True),
+        )
+        .ORDER_BY(rname.ASC(), nname.ASC())
+    )
+
+
+def sqlite_udf_gcat():
+    return regions.CALCULATE(
+        name,
+        c1=GCAT(name, "-", by=name.ASC()),
+        c2=GCAT(name, "-", by=name.DESC()),
+        c3=GCAT(name, "-", by=name.ASC(), cumulative=True),
+    ).ORDER_BY(name.ASC())
+
+
+def sqlite_udf_relmin():
+    return (
+        orders.WHERE((YEAR(order_date) == 1994) & (order_priority == "1-URGENT"))
+        .CALCULATE(month=MONTH(order_date))
+        .PARTITION(name="months", by=month)
+        .CALCULATE(
+            month,
+            n_orders=COUNT(orders),
+            m1=RELMIN(COUNT(orders)),
+            m2=RELMIN(COUNT(orders), by=month, cumulative=True),
+            m3=RELMIN(COUNT(orders), by=month, frame=(-1, 1)),
+        )
+        .ORDER_BY(month.ASC())
+    )
+
+
+def sqlite_udf_cumulative_distribution():
+    return (
+        orders.WHERE(
+            (YEAR(order_date) == 1998)
+            | ((order_priority == "2-HIGH") & (YEAR(order_date) == 1992))
+        )
+        .CALCULATE(segment=customer.market_segment)
+        .PARTITION(name="segments", by=segment)
+        .orders.CALCULATE(c=ROUND(CUMULATIVE_DISTRIBUTION(by=order_priority.ASC()), 4))
+        .PARTITION(name="groups", by=c)
+        .CALCULATE(c, n=COUNT(orders))
+        .ORDER_BY(c)
+    )
+
+
+def sqlite_udf_decode3():
+    return (
+        orders.WHERE(clerk == "Clerk#000000951")
+        .CALCULATE(
+            key,
+            DECODE3(INTEGER(order_priority[:1]), 1, "A", 2, "B", 3, "C", "D"),
+        )
+        .TOP_K(10, by=key.ASC())
+    )
```

Comment:
nit:
- More deep nested examples `ROUND(PERCENTAGE(POSITIVE(EPSILON(...))))`
- Boolean Output Used in WHERE, e..g `orders.WHERE(POSITIVE(total_price)).CALCULATE(key)`